sdk/src/wallet/migration/migrate_0.rs

```diff
@@ -169,39 +136,258 @@ trait Convert {
     fn convert(old: Self::Old) -> crate::wallet::Result<Self::New>;
 }
 
+mod types {
+    use core::str::FromStr;
+
+    use serde::{Deserialize, Serialize};
+
+    use crate::types::block::Error;
+
+    macro_rules! string_serde_impl {
+        ($type:ty) => {
+            impl serde::Serialize for $type {
+                fn serialize<S: serde::Serializer>(&self, s: S) -> Result<S::Ok, S::Error> {
+                    use alloc::string::ToString;
+
+                    s.serialize_str(&self.to_string())
+                }
+            }
+
+            impl<'de> serde::Deserialize<'de> for $type {
+                fn deserialize<D>(deserializer: D) -> Result<$type, D::Error>
+                where
+                    D: serde::Deserializer<'de>,
+                {
+                    struct StringVisitor;
+
+                    impl<'de> serde::de::Visitor<'de> for StringVisitor {
+                        type Value = $type;
+
+                        fn expecting(&self, formatter: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+                            formatter.write_str("a string representing the value")
+                        }
+
+                        fn visit_str<E>(self, v: &str) -> Result<Self::Value, E>
+                        where
+                            E: serde::de::Error,
+                        {
+                            let value = core::str::FromStr::from_str(v).map_err(serde::de::Error::custom)?;
+                            Ok(value)
+                        }
+                    }
+
+                    deserializer.deserialize_str(StringVisitor)
+                }
+            }
+        };
+    }
+
+    #[derive(Copy, Clone, PartialEq, Eq, Hash)]
+    pub struct TransactionId([u8; Self::LENGTH]);
+
+    impl TransactionId {
+        pub const LENGTH: usize = 32;
+    }
+
+    impl core::str::FromStr for TransactionId {
+        type Err = Error;
+
+        fn from_str(s: &str) -> Result<Self, Self::Err> {
+            Ok(Self(prefix_hex::decode(s).map_err(Error::Hex)?))
+        }
+    }
+
+    impl core::fmt::Display for TransactionId {
+        fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+            write!(f, "{}", prefix_hex::encode(self.0))
+        }
+    }
+
+    string_serde_impl!(TransactionId);
+
+    #[derive(Serialize, Deserialize)]
+    #[serde(rename_all = "camelCase")]
+    pub struct Transaction {
+        pub payload: TransactionPayload,
+        pub block_id: Option<serde_json::Value>,
```

Comment:
What's the reason to use `serde_json::Value` instead of `String` here?

Reply:
Consistency?